public/.htaccess

```diff
@@ -277,4 +277,25 @@ RewriteRule ^ %{ENV:BASE}/index.php [L]
             application/xml font/opentype image/svg+xml image/x-icon \
             text/css text/html text/plain text/x-component text/xml text/javascript
     </IfModule>
-</IfModule>
\ No newline at end of file
+</IfModule>
+
+# ----------------------------------------------------------------------
+# | Security settings                                                  |
+# ----------------------------------------------------------------------
+
+<IfModule mod_headers.c>
+
+    Header set X-Content-Type-Options nosniff
+    Header set X-Frame-Options SAMEORIGIN
+    Header always unset x-powered-by
+
+    # Content Security Policy (see https://content-security-policy.com/ for more information and examples)
+    Header set Content-Security-Policy "default-src 'self';"
```

Comment:
@alexloetscher95 Aren't we missing more declarations (e.g. `img-src`, `style-src`, `script-src`, etc.) here?

Reply:
@aarongerig nope; "the default-src directive defines the default policy for fetching resources such as JavaScript, Images, CSS, Fonts, AJAX requests, Frames, HTML5 Media."
By default all images, css, fonts are only allowed by same origin.

Example for allowing external image source:
Content-Security-Policy: default-src 'self'; img-src 'self' cdn.example.com;